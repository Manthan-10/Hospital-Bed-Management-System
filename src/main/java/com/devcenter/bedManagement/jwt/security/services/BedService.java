package com.devcenter.bedManagement.jwt.security.services;

import java.util.List;
import java.util.Optional;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.devcenter.bedManagement.jwt.models.Bed;
import com.devcenter.bedManagement.jwt.models.Room;
import com.devcenter.bedManagement.jwt.repository.BedRepository;
import com.devcenter.bedManagement.jwt.repository.RoomRepository;


/**
 * Service class
 * @author MN097789
 *
 */

@Service
public class BedService {
	
	@Autowired
	BedRepository bedRepository;
	
	@Autowired
	RoomRepository roomRepository;
	
	@Autowired
	private ModelMapper modelMapper;
	
	/**
	 * Function for adding new beds
	 * @param bed represents Bed object
	 * 	 
	 **/
	public void createBed(Bed bed) {
		 bedRepository.save(bed);
	}
	
	

	/**
	 * Function fetches all beds
	 * @return Returns List of all beds
	 */
	public List<Bed> getBeds() {
		return bedRepository.findAll(); 
	}
	
	
	/**
	 * Function fetches the details of all the beds for the given Room id 
	 * @param is id, which represents Room Id
	 * @return All Beds in that particular Room and its details
	 * 
	 */
	public List<Bed> getBedsByRoom(int id){
		
		Optional<Room> room = roomRepository.findById(id);
		List<Bed> beds = bedRepository.getBedsByRoom(this.modelMapper.map(room, Room.class));
		
		return beds;
		
	}
}
