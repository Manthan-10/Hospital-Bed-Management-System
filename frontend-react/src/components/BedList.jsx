import React, { useEffect, useState } from "react";
import BedService from "../services/bed.service";
import "./Bed.css";
import Bed from "./Bed";

/**
 * This component displays all the beds in a particular Room
 * using roomId of that Room
 * @param {rid}  RoomId
 */
const BedList = ({ rid }) => {
  const [data, setBedList] = useState([]);

  /**
   * @param {rid} RoomId of that Room
   * @return Beds for that Particular Room
   */

  useEffect(() => {
    BedService.getBedsByRid(rid)
      .then((response) => {
        console.log(response);
        setBedList(response.data);
      })
      .catch((e) => {
        console.log(e);
      });
  }, []);

  return (
    <div className="bed">
      {data.map((bed) => (
        <Bed bed={bed} key={bed.id} />
      ))}
    </div>
  );
};

export default BedList;
