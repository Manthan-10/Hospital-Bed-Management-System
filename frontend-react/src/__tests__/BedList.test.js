import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import BedList from "../components/BedList";
import Adapter from "enzyme-adapter-react-16";
import * as reactRouterDom from "react-router-dom";
import renderer from "react-test-renderer";
import BedService from "../services/bed.service";
import { render } from "@testing-library/react";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Testing for UI of Login page
 * @author MN097789
 */

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useSelector: jest.fn(),

  useNavigate: () => mockNavigate,
}));
const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

describe("BedList UI Test Suite", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BedList />);
  });

  test("Checks if it renders correctly", () => {
    const tree = renderer.create(<BedList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should check for divs", () => {
    wrapper = shallow(<BedList />);
    expect(wrapper.length).toEqual(1);
  });

  test("BedService.getBedsByRid is called", () => {
    jest.spyOn(React, "useEffect").mockImplementation((f) => f());
    jest.spyOn(BedService, "getBedsByRid");
    render(<BedList />);
    expect(BedService.getBedsByRid).toHaveBeenCalled();
  });
});
