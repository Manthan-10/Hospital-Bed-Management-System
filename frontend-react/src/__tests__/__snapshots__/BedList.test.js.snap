// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedList UI Test Suite Checks if it renders correctly 1`] = `
<div
  className="bed"
/>
`;
